### PR TITLE
Separate number of chunks from flags

### PR DESCRIPTION
**Pull request by avbelov23**
https://github.com/tempesta-tech/tempesta/pull/1151

### CODE REVIEW

File: tempesta_fw/str.h [hunk 1]
```diff
@@ -217,7 +214,8 @@ typedef struct tfwstr_t {
 	struct sk_buff	*skb;
 	unsigned long	len;
 	unsigned char	eolen;
-	unsigned int	flags;
+	unsigned int	nchunks;
+	unsigned char	flags;
```
[the remark this answers is not in the document]
Comment (ikoveshnikov) [reply tempesta_fw/str.h]: As soon as struct members will be rearranged, need to remove all custom initializers of the `TfwStr` placed all over the code. E.g. `TfwStr_string()` from `cache.c` must be replaced by `DEFINE_TFW_STR()`

[the remark this answers is not in the document]
Comment (avbelov23) [reply tempesta_fw/str.h]: This is different macros. May be `TFW_STR_FROM` or move `TfwStr_string` to `str.h`?

[the remark this answers is not in the document]
Comment (ikoveshnikov) [reply tempesta_fw/str.h]: I just wanted to show, that we have a lot initialising macroses all over the code. Some of them are designated and safe for members rearranges. But some of the, are positional, so rearrangements can lead to hard-to-catch issues. In the same time that initialising macroses simply init `TfwStr` to point on constant predefined string. 
I believe that all common initialisation macroses should be deefined as library macroses in `str.h`, and shouldn't be spread all over the code base.

**Comment by avbelov23** [comment]
DONE

File: tempesta_fw/str.h [hunk 2]
```diff
@@ -216,39 +211,30 @@ typedef struct tfwstr_t {
...
+	unsigned int	nchunks;
+	unsigned short	flags;
 	unsigned char	eolen;
-	unsigned int	flags;
 } TfwStr;
 
-#define DEFINE_TFW_STR(name, val) TfwStr name = { .data = (val), NULL, \
-						  sizeof(val) - 1, 0 }
-#define TFW_STR_FROM(s)		((TfwStr){.data = (char*)s, NULL, strlen(s)})
+#define TFW_STR_STRING(val)	((TfwStr){.data = (val), NULL, sizeof(val) - 1, 0})
```
Comment (ikoveshnikov) [inline tempesta_fw/str.h]: The initialisation looks incomplete, only first 4 are explicitly initialised. Either initialise all of them or only `.data` and `.len`. Same apply for `TFW_STR_FROM_CSTR()`

File: tempesta_fw/http_parser.c [hunk 3]
```diff
@@ -1499,7 +1497,7 @@ __req_parse_content_type(TfwHttpMsg *hm, unsigned char *data, size_t len)
 	}
 
 	__FSM_STATE(I_ContTypeParam) {
-		static const TfwStr s_boundary = TFW_STR_FROM("boundary=");
+		static const TfwStr s_boundary = TFW_STR_FROM_CSTR("boundary=");
```
[the remark this answers is not in the document]
Comment (avbelov23) [reply tempesta_fw/http_parser.c]: What?

**Comment by avbelov23** [comment]
DONE

**Pull request merged**